x/evm/keeper/state_transition.go

```diff
@@ -222,7 +219,11 @@ func (k *Keeper) ApplyMessage(evm *vm.EVM, msg core.Message, cfg *params.ChainCo
 		// should have already been checked on Ante Handler
 		return nil, stacktrace.Propagate(err, "intrinsic gas failed")
 	}
-	// should be > 0 as it is checked on Ante Handler
+	// Should check again even if it is checked on Ante Handler, because eth_call don't go through Ante Handler.
+	if msg.Gas() < intrinsicGas {
+		// Return the exact error, so eth_estimateGas can know what is going on.
+		return nil, core.ErrIntrinsicGas
```

Comment:
`Propagate` error and use an `sdk.Error` that uses the `core.ErrIntrinsicGas` error message

Reply:
done, we'll fix `eth_estimateGas` in another PR.